File: PersonalRecipe/PersonalCore/Service/InstructionService.cs
using PersonalCore.Models;
using PersonalCore.Models.Dto;
using PersonalCore.Repository.IRepository;
using PersonalCore.Service.IService;

namespace PersonalCore.Service;
public class InstructionService : IInstructionService
{
    private readonly IInstructionRepository _insRepo;
    public InstructionService(IInstructionRepository insRepo)
    {
        _insRepo = insRepo;
    }
    public async Task<Instruction> Create(InstructionDto dto)
    {
        return await _insRepo.Create(dto);
    }

    public async Task<bool> Delete(int id)
    {
        return await _insRepo.Delete(id);
    }

    public async Task<bool> Exists(int id)
    {
        return await _insRepo.Exists(id);
    }

    public async Task<bool> Exists(string content)
    {
        return await _insRepo.Exists(content);
    }

    public async Task<Instruction> Get(int id)
    {
        return await _insRepo.Get(id);
    }

    public async Task<List<Instruction>> GetAll(ListParams listParams)
    {
        return await _insRepo.GetAll(listParams);
    }

    public async Task<bool> Update(Instruction dto)
    {
        return await _insRepo.Update(dto);
    }
}

File: PersonalRecipe/PersonalClient/src/app/services/instruction.service.ts
import { Injectable } from '@angular/core';
import { AppService } from './app.service';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Instruction } from '../models/instruction.model';
import { ListParams } from '../models/list-params.model';

@Injectable({
  providedIn: 'root'
})
export class InstructionService {
  private apiUrl: string = "";
  private headers: HttpHeaders = new HttpHeaders({
    'Content-Type':  'application/json',
  })
  constructor(
    private appSvc: AppService,
    private http: HttpClient,
  ){ this.apiUrl = this.appSvc.getBaseUrl() + "instruction/" }

  get(id: number){ return this.http.get<Instruction>(this.apiUrl + `${id}/`) };
  create(c: Instruction){ return this.http.post<Instruction>(this.apiUrl, c) };
  update(c: Instruction){ return this.http.put(this.apiUrl + `${c.id}/`, c) };
  delete(id: number){ return this.http.get<Instruction>(this.apiUrl + `${id}/`) };
  getAll(params: ListParams){
    const headers = this.headers;
    headers.append("MultiSearch", params.multiSearch ? "true" : "false");
    headers.append("SearchString", params.searchString ?? "");
    headers.append("SearchField", params.searchField ?? "");
    headers.append("OrderBy", params.orderBy ?? "");
    headers.append("Limit", params.limit ? params.limit.toString() : "");
    return this.http.get<Instruction[]>(this.apiUrl, {...headers});
  }
}
